cellengine/resources/gate.py

```diff
@@ -126,10 +127,25 @@ def update_gate_family(experiment_id: str, gid: str, body: Dict) -> None:
         if res["nModified"] < 1:
             raise Warning("No gates updated.")
 
-    def tailor_to(self, fcs_file: FcsFile):
-        self.tailored_per_file = True
-        self.fcs_file_id = fcs_file._id
-        self.update()
+    def apply_tailoring(self, fcs_file_ids: List[str]):
+        """Tailor this gate to a specific FCS file or files."""
+        res = ce.APIClient().apply_tailoring(self.experiment_id, self, fcs_file_ids)
+        return self._make_tailored_gates(res)
+
+    def _make_tailored_gates(self, payload: TailoringDict) -> Dict[str, List[Gate]]:
+        ret = defaultdict(list)
+        for k, v in payload.items():
+            if v:
+                if k == "deleted":
+                    [ret[k].append(i["_id"]) for i in v]
+                else:
+                    [
+                        ret[k].append(
+                            ce.APIClient().get_gate(self.experiment_id, i["_id"])
```

Comment:
Making an API request for each file is going to be too slow. It's totally normal to apply tailoring to 100s of files. All of these gates are identical except for their `_id` and `fcsFileId`, so you can synthesize them locally from the `self` gate.

Reply:
I _think_ the most recent changes are what you want here